WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -140,6 +139,50 @@ public void Medication_Search(object sender, RoutedEventArgs e)
             }
         }
 
+        public void appointmentSearch_Click(object sender, RoutedEventArgs e)
+        {
+            clearAllResults();
+            if (InvalidInputForAppointment())
+            {
+                MessageBox.Show("Invalid input.");
+            }
+            else
+            {   
+                int index = doctorsComboBox.SelectedIndex;
+                DateTime startDate = DateTime.ParseExact(startDatePicker.Text + AllConstants.DayStart, "MM/dd/yyyy HH:mm", null);
+                DateTime endDate = DateTime.ParseExact(endDatePicker.Text + AllConstants.DayEnd, "MM/dd/yyyy HH:mm", null);
+                RecommendationRequestDto recommendationRequestDto = new RecommendationRequestDto()
+                {
+                    DoctorId = doctorServerController.GetDoctorsByDepartment(regularExaminationDepartment).ElementAt(index).DoctorId,
+                    SpecialtyId = regularExaminationDepartment,
+                    TimeInterval = new TimeInterval(startDate, endDate),
+                    Preference = GetRecommendationPreference()
+                };
+                if (schedulingController.GetAppointments(recommendationRequestDto) != null)
+                {
+                    appointemntSearchResult = schedulingController.GetAppointments(recommendationRequestDto);
```

Comment:
You should call controller method only one time and then check if it returns null.

Reply:
Fixed